front_end/webserver.py

```diff
@@ -760,6 +767,10 @@ def post(self, course, assignment):
             assignment_details["has_due_date"] = self.get_body_argument("has_due_date") == "Select"
             assignment_details["has_timer"] = self.get_body_argument("has_timer") == "On"
             assignment_details["enable_help_requests"] = self.get_body_argument("enable_help_requests") == "Yes"
+            assignment_details["allowed_ip_addresses"] = None
+            if self.get_body_argument("access_restricted") == "Yes":
```

Comment:
Assuming we remove the access_restricted dropdown box, you should be able to simplify this logic.